pyccel/codegen/printing/ccode.py

```diff
@@ -778,6 +782,33 @@ def _print_MathSqrt(self, expr):
             code_args = self._print(arg)
         return 'sqrt({})'.format(code_args)
 
+    # -------------- pyc_math functions ---------------------------------------
+    def _print_MathFactorial(self, expr):
+        """ Print the equivalent code to pycthon factorial function call"""
+        # include of the pyc_math header
+        self._additional_imports.add('pyc_math')
+        arg = expr.args[0]
+        return 'pyc_factorial({})'.format(self._print(arg))
+
+    def _print_MathGcd(self, expr):
+        """ Print the equivalent code to pycthon gcd function call"""
+        # include of the pyc_math header
+        self._additional_imports.add('pyc_math')
+        args = ", ".join(self._print(arg) for arg in expr.args)
+        return 'pyc_gcd({})'.format(args)
+
+    def _print_MathDegrees(self, expr):
+        """ Print the equivalent code to pycthon degrees function call"""
+        # include of the pyc_math header
+        self._additional_imports.add('pyc_math')
+        return 'pyc_degrees({})'.format(self._print(expr.args[0]))
+
+    def _print_MathRadians(self, expr):
+        """ Print the equivalent code to pycthon radians function call"""
+        # include of the pyc_math header
+        self._additional_imports.add('pyc_math')
+        return 'pyc_radians({})'.format(self._print(expr.args[0]))
+    # -------------- end of pyc_math functions --------------------------------
```

Comment:
I don't suppose all these classes with the same methods happen to inherit from a base class that would let us write one print method with a dictionary do they?

Reply:
no, they aren't.
in order to write one print for all of them, we need to create these classes:
```python
class PycMathFunction(MathFunctionBase): pass

class PycMathFunctionFloat(PycMathFunction):
    _dtype = NativeReal()
    _precision = default_precision['real']

class PycMathFunctionInt(PycMathFunction):
    _dtype = NativeInteger()
    _precision = default_precision['integer']

class PycMathFunctionBool(PycMathFunction):
    _dtype = NativeBool()
    _precision = default_precision['bool']
```
then inherit all classes for the functions that are in pyc_math lib. this seems to be useless.

Reply:
These classes already exist on my computer. But I am not clear on why `PycMathFunctionFloat` would be necessary, i.e. why you need the dtype/precision to print.

To be clear I am not suggesting the removal of the classes `MathRadians` etc, just that the printing is done from `_print_MathFunctionFloat` which would be called if we didn't have a `_print_MathRadians` function

Reply:
@rakati I don't think that we need the three intermediate classes that you wrote above. The `PycMathFunction` base class seems enough to me, and your new classes can inherit from it directly.

In the C printer you will then have just one function, called `_print_PycMathFunction`, which will print the correct function name in C using a dictionary as @EmilyBourne suggested. The body of this function will look very similar to your current `_print_MathGcd` function.

Same thing for Fortran.

Reply:
Ok, I will do that.